Gem/Code/Source/Effects/GameEffect.cpp

```diff
@@ -215,6 +215,22 @@ namespace MultiplayerSample
 #endif
     }
 
+    void GameEffect::StopEffect() const
+    {
+#if AZ_TRAIT_CLIENT
+        if (m_emitter != nullptr)
+        {
+            if (PopcornFX::PopcornFXRequests* popcornFx = PopcornFX::PopcornFXRequestBus::FindFirstHandler())
+            {
+                if (m_popcornFx->IsEffectAlive(m_emitter))
+                {
+                    m_popcornFx->DestroyEffect(m_emitter);
```

Comment:
Thanks! Fixed.

Reply:
Are you sure about that? 🤔 Wouldn't that actually leak the emitter instance, or does Stop/Kill actually delete it? The documentation (such as it is) doesn't indicate the emitter will get deleted.
Since that would effectively halt the effect from ever being started again, if what you say is true we should actually allocate a new emitter so that the effect can be triggered again, potentially in a different location.

Reply:
`	bool	PopcornFXEmitter::Stop()
	{
		if (m_Emitter != null)
		{
			m_Emitter->Stop();
			return true;
		}
		return false;
	}

	bool	PopcornFXEmitter::Terminate()
	{
		if (m_Emitter != null)
		{
			m_Emitter->Terminate();
			return true;
		}
		return false;
	}

	bool	PopcornFXEmitter::Kill()
	{
		if (m_Emitter != null)
		{
			m_Emitter->KillDeferred();
			return true;
		}
		return false;
	}`
This is as far as I can trace without code.. the instance still seems to be valid after the call. This appears to be the correct call to destroy an active instance, which you correctly identified should set the effect to nullptr after being invoked:
`		virtual void						DestroyEffect(StandaloneEmitter *emitter) = 0;
`

Reply:
agggh, sorry, my brain was mush when i wrote that. You're 100% correct, DestroyEffect() is the only one where you need to null out the emitter. The other ones keep the emitter pointer alive.